lib/pattern_query_helper/sql.rb

```diff
@@ -2,60 +2,58 @@ module PatternQueryHelper
   class Sql
     def self.sql_query(config)
       model = config[:model]
-      query = config[:query]
       query_params = config[:query_params] || {}
       page = config[:page]
       per_page = config[:per_page]
-      filter_string = config[:filter_string]
       filter_params = config[:filter_params] || {}
-      sort_string = config[:sort_string]
 
-      if page && per_page
+      if page and per_page
         query_params[:limit] = per_page
         query_params[:offset] = (page - 1) * per_page
         limit = "limit :limit offset :offset"
       end
 
+      full_count_join = "join (select count(*) as full_count from filtered_query) as filtered_query_count on true" if page or per_page
       query_params = query_params.merge(filter_params).symbolize_keys
-      sort_string = "order by #{sort_string}" if !sort_string.blank?
-      filter_string = "where #{filter_string}" if !filter_string.blank?
 
       sql = %(
-          with query as (#{query})
-          select *
-          from query
-          #{filter_string}
-          #{sort_string}
-          #{limit}
-        )
+        with filtered_query as (#{filtered_query(config)})
+        select *
+        from filtered_query
+        #{full_count_join}
+        #{limit}
+      )
 
       model.find_by_sql([sql, query_params])
     end
 
     def self.sql_query_count(config)
-      model = config[:model]
+      # not used anymore - keeping to pass existing tests
```

Comment:
Lets delete or change existing tests and remove this method to keep the code clean

Reply:
Good idea, I made those updates in commit a5a223e